pkg/schedulingtypes/plugin.go

```diff
@@ -60,6 +60,8 @@ type Plugin struct {
 	placementClient util.ResourceClient
 
 	typeConfig typeconfig.Interface
+
+	stopChannel chan struct{}
```

Comment:
Nice cleanup!  

As per a previous question, is there a reason you switched the stop channels to be bidirectional?  Most stop channels that I've seen in kube are one way, and I assume that's because the signaling is intended to only flow from caller to callee.

Reply:
we cannot close a receive-only channel. But I need to close the channel to stop related controllers.
```
$ CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o bin/hyperfed cmd/hyperfed/main.go 
\# github.com/kubernetes-sigs/federation-v2/pkg/controller/schedulingpreference
pkg/controller/schedulingpreference/controller.go:180:7: invalid operation: close(s.stopChannel) (cannot close receive-only channel)
```

Reply:
Ah, I see.  It's necessary to create the stop channel as writeable so it can be closed, and then it can be passed as read-only.  Previously the channel was only being passed in so it was always read-only.

In future, please avoid resolving conversations in which you answer a question.  That way, whoever asked the question can determine for themselves whether the response satisfied their query.